sycl/source/CMakeLists.txt

```diff
@@ -131,55 +143,33 @@ set(SYCL_SOURCES
     "$<$<OR:$<PLATFORM_ID:Linux>,$<PLATFORM_ID:Darwin>>:detail/posix_pi.cpp>"
 )
 
-if (SYCL_ENABLE_XPTI_TRACING)
-  add_sycl_rt_library(sycl ${SYCL_SOURCES} XPTI_LIB xpti)
-else()
-  add_sycl_rt_library(sycl ${SYCL_SOURCES})
-endif()
-
 if (MSVC)
   # MSVC provides two incompatible build variants for its CRT: release and debug
   # To avoid potential issues in user code we also need to provide two kinds
   # of SYCL Runtime Library for release and debug configurations.
-  set(SYCL_CXX_FLAGS "")
-  if (CMAKE_BUILD_TYPE MATCHES "Debug")
-    set(SYCL_CXX_FLAGS "${CMAKE_CXX_FLAGS_DEBUG}")
-    string(REPLACE "/MDd" "" SYCL_CXX_FLAGS "${SYCL_CXX_FLAGS}")
-    string(REPLACE "/MTd" "" SYCL_CXX_FLAGS "${SYCL_CXX_FLAGS}")
-  else()
-    if (CMAKE_BUILD_TYPE MATCHES "Release")
-      set(SYCL_CXX_FLAGS "${CMAKE_CXX_FLAGS_RELEASE}")
-    elseif (CMAKE_BUILD_TYPE MATCHES "RelWithDebInfo")
-      set(SYCL_CXX_FLAGS "${CMAKE_CXX_FLAGS_MINSIZEREL}")
-    elseif (CMAKE_BUILD_TYPE MATCHES "MinSizeRel")
-      set(SYCL_CXX_FLAGS "${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
-    endif()
-    string(REPLACE "/MD" "" SYCL_CXX_FLAGS "${SYCL_CXX_FLAGS}")
-    string(REPLACE "/MT" "" SYCL_CXX_FLAGS "${SYCL_CXX_FLAGS}")
-  endif()
-
-  # target_compile_options requires list of options, not a string
-  string(REPLACE " " ";" SYCL_CXX_FLAGS "${SYCL_CXX_FLAGS}")
-
-  set(SYCL_CXX_FLAGS_RELEASE "${SYCL_CXX_FLAGS};/MD")
-  set(SYCL_CXX_FLAGS_DEBUG "${SYCL_CXX_FLAGS};/MDd")
-
-  # CMake automatically applies these flags to all targets. To override this
-  # behavior, options lists are reset.
-  set(CMAKE_CXX_FLAGS_RELEASE "")
-  set(CMAKE_CXX_FLAGS_MINSIZEREL "")
-  set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "")
-  set(CMAKE_CXX_FLAGS_DEBUG "")
-
-  target_compile_options(sycl PRIVATE ${SYCL_CXX_FLAGS_RELEASE})
+  foreach(flag_var
+      CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_RELEASE
+      CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO)
+    string(REGEX REPLACE "/MD" "" ${flag_var} "${${flag_var}}")
+    string(REGEX REPLACE "/MT" "" ${flag_var} "${${flag_var}}")
+    string(REGEX REPLACE "/MDd" "" ${flag_var} "${${flag_var}}")
+    string(REGEX REPLACE "/MTd" "" ${flag_var} "${${flag_var}}")
+  endforeach()
 
   if (SYCL_ENABLE_XPTI_TRACING)
-    add_sycl_rt_library(sycld ${SYCL_SOURCES} XPTI_LIB xptid)
+    add_sycl_rt_library(sycld XPTI_LIB xptid COMPILE_OPTIONS "/MDd" SOURCES ${SYCL_SOURCES})
   else()
-    add_sycl_rt_library(sycld ${SYCL_SOURCES})
+    add_sycl_rt_library(sycld COMPILE_OPTIONS "/MDd" SOURCES ${SYCL_SOURCES})
   endif()
-  target_compile_options(sycld PRIVATE ${SYCL_CXX_FLAGS_DEBUG})
 
+  set(SYCL_EXTRA_OPTS "/MD")
```

Comment:
What is this for?

Reply:
The above code block effectively removes all CRT flags for sources in current directory. We need to link sycl.dll against C++ runtime lib, and thus this flag. On Linux we don't need additional flags, so we don't set anything.